xopt/base.py

```diff
@@ -154,9 +154,6 @@ def submit_data(self, input_data: pd.DataFrame):
         futures = self.evaluator.submit_data(input_data)
         self._futures.update(futures)
 
-        # wait for futures
-        self.wait_for_futures()
```

Comment:
The issue with this change is that if users want to evaluate specific data points using submit_data this will not update the internal dataframe with the data, maybe add a keyword argument to incl. this behavior?

Reply:
Alternatively we could make this method private and define a separate function for users, would also impact the TODO

Reply:
This is part of the `summit_` method, which is consistent with the meaning of `submit` in https://peps.python.org/pep-3148/ and returns (internally) futures.

Reply:
We could instead have an `evaluate_data`